### ISSUE TITLE
support for decimals without leading zeros

### ISSUE THREAD

**Issue by oboesax** [uklimaschewski/EvalEx#53]
in Expression.java, line 595, I had to change the line to:

```
if (Character.isDigit(ch) || (ch == decimalSeparator && Character.isDigit(peekNextChar()))) {
```

in order to account for .5 instead of 0.5

**Comment by uklimaschewski** [comment]
Thank you, but it doesn't work for all cases. I wrote a small test:

    @Test
    public void testNoLeadingZero() {
        Expression e = new Expression("0.1 + .1");
        assertEquals("0.2", e.eval().toPlainString());
		
        e = new Expression(".2*.3");
        assertEquals("0.06", e.eval().toPlainString());
	
        e = new Expression(".2*.3+.1");
        assertEquals("0.16", e.eval().toPlainString());
    }

The expression `.2*.3` gives an error `Unknown operator *. at position 3`

I think your request needs some more changes to work.

**Comment by uklimaschewski** [comment]
I think, we can close this. It has been implemented in version 1.17
